### PR TITLE
ES6 refactor

### PR DESCRIPTION
**Pull request by JPeer264**
(#355 )  I did some more refactoring. Now everything except `move/index`, `copy/ncp` and `remove/rimraf` are as much as possible refactored to ES6. I wasn't sure on the skipped files, as they are from another library or got a comment with something like `needs rewrite` or similar.

### CODE REVIEW

File: lib/ensure/symlink-paths.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-var path = require('path')
+'use strict'
+
+const path = require('path')
 // path.isAbsolute shim for Node.js 0.10 support
```
Comment (RyanZim) [inline lib/ensure/symlink-paths.js]: Sort of unrelated, but this comment should have been removed earlier, wouldn't hurt to remove it here.

**Comment by RyanZim** [comment]
I'm fine with doing the skipped files, all of them are already significantly modified from their original state.

If we decide to make changes in https://github.com/jprichardson/node-fs-extra/issues/292, you could skip `ncp`; not sure what we're doing there yet.

**Comment by JPeer264** [comment]
@RyanZim I am on `move/index` right now. As there are many functions inside I would suggest to outsource them into own files and require them separately into `index.js`. How you think about it?

**Comment by RyanZim** [comment]
`move/index` is only 158 lines, I wouldn't split it personally, perhaps @jprichardson differs.

BTW, don't bother with `copy/ncp`; I'm working on that right now.

**Comment by manidlou** [comment]
@RyanZim is this ready to merge?

**Comment by RyanZim** [comment]
Merged. @JPeer264 Thanks again for all your work here.

**Pull request merged**